c/tskit/trees.c

```diff
@@ -2558,14 +2738,46 @@ sample_sets_to_bit_array(const tsk_treeseq_t *self, const tsk_size_t *sample_set
     return ret;
 }
 
+static int
+check_sites(const tsk_id_t *sites, tsk_size_t num_sites, tsk_size_t num_site_rows)
+{
+    int ret = 0;
+    tsk_id_t id;
+    tsk_size_t i;
+
+    if (sites == NULL) {
+        goto out;
+    }
+
```

Comment:
I'd like to add a few more error values, but I assumed that would merit some discussion.

Something like: `TSK_ERR_DUPLICATE_SITES`, and `TSK_ERR_NULL_SITE`.

Reply:
Let's file a follow-up issue on this one - additional error cases for 2 site stats.